packages/react-aria/src/hooks/useAriaButton.test.ts

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import { useARIAButton } from './useAriaButton';
+import { renderHook } from '@testing-library/react-hooks';
+import { ObjectShorthandProps } from '@fluentui/react-utilities';
+
+describe('useARIAButton', () => {
+  it('should return by default shorthand props for a button', () => {
```

Comment:
chore: can you pls add test case for situation when `as` is specified as non string to properly test the `else` branch of main logic ?

Reply:
I didn't implement that use case since [RFC as prop](https://github.com/microsoft/fluentui/blob/master/rfcs/convergence/as-prop.md) has already been approved, meaning that `as` props will only accept `keyof JSX.IntrinsicElements`, that else branch should actually eventually be removed.

Reply:
gotcha, that was confusing for sure, lets remove that else then. thanks for explanation 🙌